flink/v1.17/flink/src/test/java/org/apache/iceberg/flink/sink/TestFlinkIcebergSink.java

```diff
@@ -369,4 +369,25 @@ public void testOverrideWriteConfigWithUnknownFileFormat() {
         .isInstanceOf(IllegalArgumentException.class)
         .hasMessage("Invalid file format: UNRECOGNIZED");
   }
+
+  @Test
+  public void testWriteRowReloadEnabled() throws Exception {
+    List<Row> rows = Lists.newArrayList(Row.of(1, "hello"), Row.of(2, "world"), Row.of(3, "foo"));
+    DataStream<RowData> dataStream =
+        env.addSource(createBoundedSource(rows), ROW_TYPE_INFO)
+            .map(CONVERTER::toInternal, FlinkCompatibilityUtil.toTypeInfo(SimpleDataUtil.ROW_TYPE));
+
+    FlinkSink.forRowData(dataStream)
+        .table(table)
+        .tableLoader(tableLoader)
+        .writeParallelism(parallelism)
+        .reloadIntervalMs(1000)
```

Comment:
is this testing with a reload interval just to make sure nothing is failing? or would the behavior be different when table reloading would be disabled?

Reply:
If I understand right, internally a reloading table supplier will be created which actually performs the periodic load table. I guess this test is just making sure everything works as expected when this is specified. If there's a simple way for passing some sort of mocked table loader and verifying it gets called multiple times that's a stronger test

Reply:
Right, this was more of a sanity test to ensure the sink is created when the reload interval is specified.

Reply:
ok thanks for clarifying, that's how I understood the test as well.